src/HtmlProcessor/HtmlParser.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\HtmlProcessor;
```

Comment:
I think this should belong in the `Pelago\Emogrifier\Utilities` namespace (and probably marked `@internal`).